Issue with SSL (tlsv1.3) communication in OTP 25 and OTP 26 compared to OTP 24

**Describe the bug**
I encountered an issue when using the `:ssl` with `tlsv1.3` module for SSL communication in Erlang/OTP 25 and OTP 26 compared to OTP 24.

**To Reproduce**

I used the following Erlang code to fetch data using SSL:

```erlang
-module(ssl_debug).
-export([run/0]).

run() ->
    SslOpts = [
        {verify, verify_peer},
        {cacertfile, "/path/to/ca.crt"},
        {certfile, "/path/to/client.crt"},
        {keyfile, "/path/to/client.key"},
        {keep_secrets, true},
        {versions, ['tlsv1.3']},
        {log_level, debug},
        {log_alert, true}
    ],
    Host = "mywebserver.com",
    Port = 443,
    io:format("Attempting SSL connection to ~s:~p~n", [Host, Port]),
    ssl:start(),
    case ssl:connect(Host, Port, SslOpts, 5000) of
        {ok, Socket} ->
            io:format("SSL connection established~n"),
            Path = "/myendpoint",
            Request = "GET " ++ Path ++ " HTTP/1.1\r\nHost: " ++ Host ++ "\r\n\r\n",
            case ssl:send(Socket, Request) of
                ok ->
                    io:format("Request sent~n");
                {error, SendReason} ->
                    io:format("Error sending request: ~p~n", [SendReason])
            end,
            io:format("Attempting SSL connection to ~s:~p~n", [Host, Port]),
            case receive_response(Socket, [], 3000) of
                {ok, Data} ->
                    io:format("Response:~n~s~n", [Data]);
                {error, ReceiveReason} ->
                    io:format("Error receiving response: ~p~n", [ReceiveReason])
            end,
            ok = ssl:close(Socket);
        {error, ConnectReason} ->
            io:format("SSL connection failed1: ~p~n", [ConnectReason]);
        Everything ->
            io:format("SSL connection failed2: ~p~n", [Everything])
    end.

receive_response(Socket, Acc, Timeout) ->
    receive
        {ssl, Socket, Data} ->
            io:format("Received data: ~p~n", [Data]),
            receive_response(Socket, [Acc, Data], Timeout);
        {ssl_closed, Socket} ->
            io:format("Socket closed~n"),
            {ok, iolist_to_binary(Acc)};
        {ssl_error, Socket, Reason} ->
            io:format("SSL error: ~p~n", [Reason]),
            {error, Reason};
        Everything ->
            io:format("SSL error: ~p~n", [Everything]),
            {error, "Error"}
    after Timeout ->
        io:format("Receive timeout after ~p milliseconds~n", [Timeout]),
        {ok, iolist_to_binary(Acc)}
    end.

```

The code worked perfectly on OTP 24, but it failed on both OTP 25 and OTP 26.

**Expected behavior**
I would expect to be able to connect to myserver on both OTP 25 and OTP 26.

**Affected versions**
**Erlang/OTP 26 [erts-14.2.2]** and **Erlang/OTP 25**

**Additional context**
If I use `:ssl` with `tlsv1.2` everything works on **Erlang/OTP 25** and **Erlang/OTP 25**
The problem seems to be with the `ssl_app` version. It changed from `{ssl_app,"10.7.3.5"` in **OTP 24** to `{ssl_app,"11.1.1"}` in OTP 26.
I also create this issue in elixir MINT library [here](https://github.com/elixir-mint/mint/issues/427).
Could you please investigate this issue and provide guidance on how to resolve it? Is there any workaround or fix available to ensure compatibility with the latest OTP releases?

@IngelaAndin thank you! 
It seems the last commit (e.g. `2d91ab8c02209b7f8bf699d427e7ec27722289dc) has compilation error
```bash
❯ asdf install erlang ref:2d91ab8c02209b7f8bf699d427e7ec27722289dc
asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc is not a kerl-managed Erlang/OTP installation
The asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc build has been deleted
Checking out Erlang/OTP git repository from https://github.com/erlang/otp.git...
Building Erlang/OTP 2d91ab8c02209b7f8bf699d427e7ec27722289dc from git, please wait...
APPLICATIONS DISABLED (See:  /path/to/.asdf/plugins/erlang/kerl-home/builds/asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc/otp_build_git.log)
 * jinterface     : No Java compiler found
 * odbc           : ODBC library - link check failed

APPLICATIONS INFORMATION (See:  /path/to/.asdf/plugins/erlang/kerl-home/builds/asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc/otp_build_git.log)
 * wx             : Can not link wx program are all developer packages installed?

Build failed.
compile: warnings being treated as errors
ssl_certificate.erl:318:44: variable 'DerCerts' is unused
%  318|         {ok, {_, [Cert | _] = EChain}, {_, DerCerts}}  ->
%     |                                            ^

make[3]: *** [../ebin/ssl_certificate.beam] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [opt] Error 2
make[1]: *** [opt] Error 2
make: *** [libs] Error 2

Please see /path/to/.asdf/plugins/erlang/kerl-home/builds/asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc/otp_build_git.log for full details.
Removing all artifacts except the logfile
(Use KERL_AUTOCLEAN=0 to keep build on failure, if desired)
Cleaning up compilation products for asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc
Cleaned up compilation products for asdf_2d91ab8c02209b7f8bf699d427e7ec27722289dc under /Users/dimova01/.asdf/plugins/erlang/kerl-home/builds
```

--- a/lib/ssl/src/ssl_certificate.erl
+++ b/lib/ssl/src/ssl_certificate.erl
@@ -315,8 +315,8 @@ handle_cert_auths(Chain, [], _, _) ->
     {ok, Chain};
 handle_cert_auths([Cert], CertAuths, CertDbHandle, CertDbRef) ->
     case certificate_chain(Cert, CertDbHandle, CertDbRef, [], both) of
-        {ok, {_, [Cert | _] = EChain}, {_, [_ | DCerts]}}  ->
-            case cert_auth_member(cert_issuers(DCerts), CertAuths) of
+        {ok, {_, [Cert | _] = EChain}, _}  ->
+            case cert_auth_member(cert_issuers(EChain), CertAuths) of
                 true ->
                     {ok, EChain};
                 false ->
@@ -325,8 +325,8 @@ handle_cert_auths([Cert], CertAuths, CertDbHandle, CertDbRef) ->
         _ ->
             {ok, [Cert]}
     end;
-handle_cert_auths([_ | Certs] = EChain, CertAuths, _, _) ->
-    case cert_auth_member(cert_issuers(Certs), CertAuths) of
+handle_cert_auths([_ | _] = EChain, CertAuths, _, _) ->
+    case cert_auth_member(cert_issuers(EChain), CertAuths) of
         true ->
             {ok, EChain};
         false ->

--- a/lib/ssl/test/ssl_cert_SUITE.erl
+++ b/lib/ssl/test/ssl_cert_SUITE.erl
@@ -125,7 +125,9 @@
          signature_algorithms_bad_curve_secp521r1/0,
          signature_algorithms_bad_curve_secp521r1/1,
          server_certificate_authorities_disabled/0,
-         server_certificate_authorities_disabled/1
+         server_certificate_authorities_disabled/1,
+         cert_auth_in_first_ca/0,
+         cert_auth_in_first_ca/1
          ]).
 
 %%--------------------------------------------------------------------
@@ -191,6 +193,7 @@ tls_1_3_tests() ->
      hello_retry_request,
      custom_groups,
      client_auth_no_suitable_chain,
+     cert_auth_in_first_ca,
      hello_retry_client_auth,
      hello_retry_client_auth_empty_cert_accepted,
      hello_retry_client_auth_empty_cert_rejected,
@@ -981,6 +984,25 @@ key_auth_ext_sign_only(Config) when is_list(Config) ->
     ssl_test_lib:basic_test(ClientOpts, ServerOpts, Config).
 
 %%--------------------------------------------------------------------
+cert_auth_in_first_ca() ->
+    [{doc,"Test cert auth will be available in first ca in chain, make it happen by only having one"}].
+cert_auth_in_first_ca(Config) when is_list(Config) ->
+    #{server_config := ServerOpts0,
+      client_config := ClientOpts0} =
+        public_key:pkix_test_data(#{server_chain => #{root => [{key, ssl_test_lib:hardcode_rsa_key(1)}],
+                                                      intermediates => [[]],
+                                                      peer => [{key, ssl_test_lib:hardcode_rsa_key(5)}]},
+                                    client_chain => #{root => [{key, ssl_test_lib:hardcode_rsa_key(3)}], 
+                                                      intermediates => [[]],
+                                                      peer => [{key, ssl_test_lib:hardcode_rsa_key(1)}]}}), 
+    ClientOpts = [{verify, verify_peer} | ssl_test_lib:ssl_options(extra_client, client_cert_opts, Config)],
+    ServerOpts =  [{verify, verify_peer} | ssl_test_lib:ssl_options(extra_server, server_cert_opts, Config)],
+
+    ssl_test_lib:basic_test(ClientOpts, ServerOpts, Config).
+
+%%--------------------------------------------------------------------
+
+
 longer_chain() ->
     [{doc,"Test depth option"}].
 longer_chain(Config) when is_list(Config) ->      

